src/schema/nmdc.yaml

```diff
@@ -1863,19 +1864,14 @@ classes:
       - target_subfragment
 
     slot_usage:
-      id:
-        required: true
-        structured_pattern:
-          syntax: "{id_nmdc_prefix}:(datgen|omprc)-{id_shoulder}-{id_blade}{id_version}{id_locus}"
-          interpolated: true
       has_input:
         required: true
         pattern: "^nmdc:(bsm|procsm)-[0-9][a-z]{0,6}[0-9]-[A-Za-z0-9]{1,}(\\.[A-Za-z0-9]{1,})*(_[A-Za-z0-9_\\.-]+)?$"
         comments:
           - pattern should allow typecode for Biosample and ProcessedSample
       part_of:
         range: DataGeneration
-        pattern: "^nmdc:datgen-[0-9][a-z]{0,6}[0-9]-[A-Za-z0-9]{1,}(\\.[A-Za-z0-9]{1,})*(_[A-Za-z0-9_\\.-]+)?$" # better applied as a structured_pattern, but even that might get out of sync from the asserted Study structured_pattern
```

Comment:
is this the pattern for a `DataGeneration`'s `id`? if `DataGeneration` is `abstract`, then it can't be instantiated. So an `id` pattern is irrelevant. In that case, we only have to worry about asserting the patterns for the subclasses.

I'm starting to think that we don't really want `DataGeneration` to be `abstract`